src/cache/lru-cache.ts

```diff
@@ -6,21 +6,52 @@
  * The cache is implemented as a Map, which is a built-in JavaScript object.
```

Comment:
WRT directory naming - it is a cache. Are you imagining this a generic LRU cache now and no longer directly tied to assignment logging? It does seem that way!